src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -331,14 +331,19 @@ public void fireFileStarted(String fileName) {
     @Override
     public void fireErrors(String fileName, SortedSet<LocalizedMessage> errors) {
         final String stripped = CommonUtils.relativizeAndNormalizePath(basedir, fileName);
+        boolean removeCache = false;
         for (final LocalizedMessage element : errors) {
             final AuditEvent event = new AuditEvent(this, stripped, element);
             if (filters.accept(event)) {
+                removeCache = true;
                 for (final AuditListener listener : listeners) {
                     listener.addError(event);
                 }
             }
         }
+        if (removeCache && cache != null) {
+            cache.remove(fileName);
+        }
```

Comment:
ok, I fed up with that fireError. One day we will redo this - #3538.
I am ok with your change for now.